webui/src/pages/auth/login.tsx

```diff
@@ -98,29 +110,78 @@ const LoginForm = ({loginConfig}: {loginConfig: LoginConfig}) => {
     )
 }
 
+/**
+ * Decides whether to use the login method selection flow - render the login selection component or fall back to the
+ * local lakeFS login form or the legacy flow handled by the caller.
+ *
+ * Flow:
```

Comment:
Well, that's an improvement than having no documentation at all.

However, documenting using `if - else` or `plugin returns null` is still pretty confusing to the reader.

When doing so, it either means that you can describe it better in terms of business logic / product behavior, or that the code can be simplified to a level where it needs no documentation.